scripts/codemods/kwarg_permissions.py

```diff
@@ -0,0 +1,158 @@
+
+        # remove all of the existing permissions from the params
+        def remove_existing_permissions(params: cst.Parameters) -> cst.Parameters:
+
+            existing_params = list(params.params)
+            for param in existing_params.copy():
+                if param.name.value in self.permissions:
+                    existing_params.remove(param)
+
+            # unlike params, these may contain generated objects
+            # we only have to do this for overloads, as we only change overloads directly
+            existing_kwonly_params = list(params.kwonly_params)
+            if is_overload:
+                # make matches
+                found_start = False
+                for param in existing_kwonly_params.copy():
+                    if m.matches(param.name, perm_matchers):
+                        found_start = True
+                    if found_start:
+                        existing_kwonly_params.remove(param)
+            else:
+                for param in existing_kwonly_params.copy():
+                    if m.matches(param.name, perm_matchers):
+                        existing_kwonly_params.remove(param)
+
+            star_arg = params.star_arg if existing_kwonly_params else cst.MaybeSentinel.DEFAULT
+            return params.with_changes(
+                params=existing_params,
+                kwonly_params=existing_kwonly_params,
+                star_arg=star_arg,
+            )
+
+        def get_perm_kwargs():
+            return [
+                cst.Param(
+                    cst.Name(perm),
+                    annotation,
+                    default=cst.Ellipsis(),
+                )
+                for perm in self.permissions
+            ]
+
+        # get a Params with all of the new params that we should have
+        params = remove_existing_permissions(node.params)
+        params = params.with_changes(star_kwarg=None)
+        empty_overload_params = params.deep_clone()
+
+        # add the permissions to the kw_only params
+        kwonly_params = list(params.kwonly_params)
+        kwonly_params.extend(get_perm_kwargs())
+        params = params.with_changes(kwonly_params=kwonly_params)
+
+        if is_overload:
+            node = node.with_changes(params=params)
+            return node
+
+        # make an overload before permissions
+        empty_overload = node.deep_clone().with_changes(params=empty_overload_params)
+        empty_overload = empty_overload.with_changes(
+            body=cst.IndentedBlock([cst.SimpleStatementLine([cst.Expr(cst.Ellipsis())])]),
+            decorators=[
+                cst.Decorator(cst.Name("overload")),
+                cst.Decorator(cst.Name("_generated")),
+            ],
+        )
+        # if the decorated method is an overload we make an in-place change and don't add overloads
+
+        overload = empty_overload.deep_clone()
+        overload = overload.with_changes(params=params)
```

Comment:
```suggestion
        overload = empty_overload.deep_clone().with_changes(params=params)
```
`empty_overload` is created in a similar way

Reply:
https://github.com/DisnakeDev/disnake/pull/708/commits/3bee9c80c01656c72e78484bcb00d99c8481dc85